### DISCUSSION TITLE
ISR getStaticProps get the current request host

### DISCUSSION THREAD

**Discussion by Sylvenas** [vercel/next.js#32205] [Help]
### Describe the feature you'd like to request

`getStaticProps` is not static in the `ISR` scenario, at runtime, it will still be called, at this time, I need to know the current host, a very common scenario:

Test environment server (page url: `https://test.xxx.com/data/1`), I need to call `test.xxx.com/api/data/list`.

And in the online environment (page url: `https://xxx.com/data/1`) I need to call `prod.xxx. com/api/data/list`.

If I can't get the host in `getStaticProps`, I won't be able to distinguish between the two environments.

### Describe the solution you'd like

``` js
async getStaticProps(context){
 const { request } = context;
 const { hostname } = request
 const data = await fetch(hostname + '/api/data')
 return {
   props: data,
   revalidate: 10,
 }
}
```

This way I can request different api's in different deployment environments.

### Describe alternatives you've considered

I know that setting different environment variables on different servers might solve this problem,

``` hs
// Production
URL="https://prod.xxx.com
```

but it would result in me having two sets of code, or two sets of envs, which could easily lead to BUG, and I can't guarantee that everyone, before deploying an online environment, will set the envs correctly.

**Comment by adamrights** [comment]
With ISR, the page will be served from the ISR cache without calling `getStaticProps`.

Only `amp` and `locale` can be used to "vary" the cache key, so you would potentially be serving the page built with props from API called for host A when host B is the incoming request.

You could use SSR with a reverse caching proxy that uses `host` as a key. But in your case I think an environment variable is the proper solution.

**Comment by balazsorban44** [comment]
You could ensure though, just add a script/check in your build pipeline making sure that env var is set.

If you are on Vercel for example, you already have `VERCEL_URL`, which you can be sure about to always exist.
